std/process.d

```diff
@@ -2219,6 +2239,52 @@ int wait(Pid pid) @safe
     else version (Posix) assert(pid.osHandle < 0);
 }
 
+/**
+Waits until either the process associated with `pid` terminates or the
+elapsed time exceeds the given timeout.
+
+If the process terminates within the given duration it behaves exactly like
+`wait`, except that it returns a tuple `(true, exit code)`.
+
+If the process does not terminate within the given duration it will stop
+waiting and return `(false, 0).`
+
+The timeout may not exceed `(uint.max - 1).msecs` (~ 7 weeks, 17 hours).
+
+Returns:
+An $(D std.typecons.Tuple!(bool, "terminated", int, "status")).
+
+Throws:
+$(LREF ProcessException) on failure or on attempt to wait for detached process.
+
+Example:
+See the $(LREF spawnProcess) documentation.
+
+See_also:
+$(LREF wait), for a blocking function without timeout.
+$(LREF tryWait), for a non-blocking function without timeout.
+*/
+version (Windows)
+auto waitTimeout(Pid pid, Duration timeout) @safe
+{
+    import std.typecons : Tuple;
+    assert(pid !is null, "Called wait on a null Pid.");
+    auto code = pid.performWait(timeout);
+    return Tuple!(bool, "terminated", int, "status")(pid._processID == Pid.terminated, code);
+}
+
+version (Windows)
+@system unittest // Pid and wait()
+{
+    import std.typecons : tuple;
+    TestScript prog = "timeout /T %~1";
+    auto pid = spawnProcess([prog.path, "0"]);
+    assert(waitTimeout(pid, 1.msecs) == tuple(true, 0));
```

Comment:
`timeout` docs say the argument takes seconds:
https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/timeout_1

Why are the test durations in milliseconds?

Reply:
Fixed